openblas 0.3.12 x86_64 build fails on RHEL7: dasum_k_SKYLAKEX: incompatible type

`openblas@0.3.12 target=x86_64` fails to build for `RHEL 7` on `x86_64` using:
* `GCC 8.3.1`
* `spack@develop`
* 3536433c41fa67b7ad006ab0c8787f91b3294d58
* `Sun Nov 15 11:54:39 2020 -0600`

Concrete spec: [openblas.spec.yaml.txt](https://github.com/spack/spack/files/5543207/openblas.spec.yaml.txt)
Full build log: [openblas-build-out.txt](https://github.com/spack/spack/files/5543215/openblas-build-out.txt)

```
$> spack install -f ./openblas.spec.yaml
...
6 errors found in build log:
     7200    /opt/spack/lib/spack/env/gcc/gcc -c -O2 -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DDYNAMIC_ARCH -DNO_WARMUP -DMAX_CPU_NUMBER=72 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SING
             LE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.12\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=dtrmm_iunncopy_SKYLAKEX -DASMFN
             AME=dtrmm_iunncopy_SKYLAKEX_ -DNAME=dtrmm_iunncopy_SKYLAKEX_ -DCNAME=dtrmm_iunncopy_SKYLAKEX -DCHAR_NAME=\"dtrmm_iunncopy_SKYLAKEX_\" -DCHAR_CNAME=\"dtrmm_iunncopy_SKYLAKEX\" -DNO_AFF
             INITY -DTS=_SKYLAKEX -I.. -DBUILD_KERNEL -DTABLE_NAME=gotoblas_SKYLAKEX -march=skylake-avx512 -DDOUBLE  -UCOMPLEX -Wno-uninitialized -DDOUBLE -UCOMPLEX -UOUTER -ULOWER -UUNIT generic/
             trmm_uncopy_16.c -o dtrmm_iunncopy_SKYLAKEX.o
     7201    /opt/spack/lib/spack/env/gcc/gcc -c -O2 -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DDYNAMIC_ARCH -DNO_WARMUP -DMAX_CPU_NUMBER=72 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SING
             LE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.12\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=dtrmm_ilnucopy_SKYLAKEX -DASMFN
             AME=dtrmm_ilnucopy_SKYLAKEX_ -DNAME=dtrmm_ilnucopy_SKYLAKEX_ -DCNAME=dtrmm_ilnucopy_SKYLAKEX -DCHAR_NAME=\"dtrmm_ilnucopy_SKYLAKEX_\" -DCHAR_CNAME=\"dtrmm_ilnucopy_SKYLAKEX\" -DNO_AFF
             INITY -DTS=_SKYLAKEX -I.. -DBUILD_KERNEL -DTABLE_NAME=gotoblas_SKYLAKEX -march=skylake-avx512 -DDOUBLE  -UCOMPLEX -Wno-uninitialized -DDOUBLE -UCOMPLEX -UOUTER -DLOWER -DUNIT generic/
             trmm_lncopy_16.c -o dtrmm_ilnucopy_SKYLAKEX.o
     7202    /opt/spack/lib/spack/env/gcc/gcc -c -O2 -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DDYNAMIC_ARCH -DNO_WARMUP -DMAX_CPU_NUMBER=72 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SING
             LE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.12\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=dtrmm_ilnncopy_SKYLAKEX -DASMFN
             AME=dtrmm_ilnncopy_SKYLAKEX_ -DNAME=dtrmm_ilnncopy_SKYLAKEX_ -DCNAME=dtrmm_ilnncopy_SKYLAKEX -DCHAR_NAME=\"dtrmm_ilnncopy_SKYLAKEX_\" -DCHAR_CNAME=\"dtrmm_ilnncopy_SKYLAKEX\" -DNO_AFF
             INITY -DTS=_SKYLAKEX -I.. -DBUILD_KERNEL -DTABLE_NAME=gotoblas_SKYLAKEX -march=skylake-avx512 -DDOUBLE  -UCOMPLEX -Wno-uninitialized -DDOUBLE -UCOMPLEX -UOUTER -DLOWER -UUNIT generic/
             trmm_lncopy_16.c -o dtrmm_ilnncopy_SKYLAKEX.o
     7203    /opt/spack/lib/spack/env/gcc/gcc -c -O2 -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DDYNAMIC_ARCH -DNO_WARMUP -DMAX_CPU_NUMBER=72 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SING
             LE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.12\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=dtrmm_iutucopy_SKYLAKEX -DASMFN
             AME=dtrmm_iutucopy_SKYLAKEX_ -DNAME=dtrmm_iutucopy_SKYLAKEX_ -DCNAME=dtrmm_iutucopy_SKYLAKEX -DCHAR_NAME=\"dtrmm_iutucopy_SKYLAKEX_\" -DCHAR_CNAME=\"dtrmm_iutucopy_SKYLAKEX\" -DNO_AFF
             INITY -DTS=_SKYLAKEX -I.. -DBUILD_KERNEL -DTABLE_NAME=gotoblas_SKYLAKEX -march=skylake-avx512 -DDOUBLE  -UCOMPLEX -Wno-uninitialized -DDOUBLE -UCOMPLEX -UOUTER -ULOWER -DUNIT generic/
             trmm_utcopy_16.c -o dtrmm_iutucopy_SKYLAKEX.o
     7204    In file included from ../kernel/x86_64/dasum.c:8:
     7205    ../kernel/x86_64/dasum_microk_skylakex-2.c: In function 'dasum_kernel':
  >> 7206    ../kernel/x86_64/dasum_microk_skylakex-2.c:42:38: error: incompatible type for argument 1 of '_mm512_abs_pd'
     7207                 accum_0 += _mm512_abs_pd(_mm512_load_pd(&x1[i + 0]));
     7208                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~
     7209    In file included from /opt/software/linux-rhel7-x86_64/gcc-4.8.5/gcc-8.1.0-eesg3q2ggeqhvjpopgbpobxgyv6m5hg4/lib/gcc/x86_64-pc-linux-gnu/8.1.0/include/immintrin.h:45,
     7210                     from ../kernel/x86_64/dasum_microk_skylakex-2.c:6,
     7211                     from ../kernel/x86_64/dasum.c:8:
     7212    /opt/software/linux-rhel7-x86_64/gcc-4.8.5/gcc-8.1.0-eesg3q2ggeqhvjpopgbpobxgyv6m5hg4/lib/gcc/x86_64-pc-linux-gnu/8.1.0/include/avx512fintrin.h:7730:23: note: expected '__m512' {aka '
             __vector(16) float'} but argument is of type '__m512d' {aka '__vector(8) double'}
     7213     _mm512_abs_pd (__m512 __A)
     7214                    ~~~~~~~^~~
     7215    In file included from ../kernel/x86_64/dasum.c:8:
  >> 7216    ../kernel/x86_64/dasum_microk_skylakex-2.c:43:38: error: incompatible type for argument 1 of '_mm512_abs_pd'
     7217                 accum_1 += _mm512_abs_pd(_mm512_load_pd(&x1[i + 8]));
     7218                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~
     7219    In file included from /opt/software/linux-rhel7-x86_64/gcc-4.8.5/gcc-8.1.0-eesg3q2ggeqhvjpopgbpobxgyv6m5hg4/lib/gcc/x86_64-pc-linux-gnu/8.1.0/include/immintrin.h:45,
     7220                     from ../kernel/x86_64/dasum_microk_skylakex-2.c:6,
     7221                     from ../kernel/x86_64/dasum.c:8:
     7222    /opt/software/linux-rhel7-x86_64/gcc-4.8.5/gcc-8.1.0-eesg3q2ggeqhvjpopgbpobxgyv6m5hg4/lib/gcc/x86_64-pc-linux-gnu/8.1.0/include/avx512fintrin.h:7730:23: note: expected '__m512' {aka '
             __vector(16) float'} but argument is of type '__m512d' {aka '__vector(8) double'}
     7223     _mm512_abs_pd (__m512 __A)
     7224                    ~~~~~~~^~~
     7225    In file included from ../kernel/x86_64/dasum.c:8:
  >> 7226    ../kernel/x86_64/dasum_microk_skylakex-2.c:44:38: error: incompatible type for argument 1 of '_mm512_abs_pd'
     7227                 accum_2 += _mm512_abs_pd(_mm512_load_pd(&x1[i +16]));
     7228                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~
     7229    In file included from /opt/software/linux-rhel7-x86_64/gcc-4.8.5/gcc-8.1.0-eesg3q2ggeqhvjpopgbpobxgyv6m5hg4/lib/gcc/x86_64-pc-linux-gnu/8.1.0/include/immintrin.h:45,
     7230                     from ../kernel/x86_64/dasum_microk_skylakex-2.c:6,
     7231                     from ../kernel/x86_64/dasum.c:8:
     7232    /opt/software/linux-rhel7-x86_64/gcc-4.8.5/gcc-8.1.0-eesg3q2ggeqhvjpopgbpobxgyv6m5hg4/lib/gcc/x86_64-pc-linux-gnu/8.1.0/include/avx512fintrin.h:7730:23: note: expected '__m512' {aka '
             __vector(16) float'} but argument is of type '__m512d' {aka '__vector(8) double'}
     7233     _mm512_abs_pd (__m512 __A)
     7234                    ~~~~~~~^~~
     7235    In file included from ../kernel/x86_64/dasum.c:8:
  >> 7236    ../kernel/x86_64/dasum_microk_skylakex-2.c:45:38: error: incompatible type for argument 1 of '_mm512_abs_pd'
     7237                 accum_3 += _mm512_abs_pd(_mm512_load_pd(&x1[i +24]));
     7238                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~
     7239    In file included from /opt/software/linux-rhel7-x86_64/gcc-4.8.5/gcc-8.1.0-eesg3q2ggeqhvjpopgbpobxgyv6m5hg4/lib/gcc/x86_64-pc-linux-gnu/8.1.0/include/immintrin.h:45,
     7240                     from ../kernel/x86_64/dasum_microk_skylakex-2.c:6,
     7241                     from ../kernel/x86_64/dasum.c:8:
     7242    /opt/software/linux-rhel7-x86_64/gcc-4.8.5/gcc-8.1.0-eesg3q2ggeqhvjpopgbpobxgyv6m5hg4/lib/gcc/x86_64-pc-linux-gnu/8.1.0/include/avx512fintrin.h:7730:23: note: expected '__m512' {aka '
             __vector(16) float'} but argument is of type '__m512d' {aka '__vector(8) double'}

     ...

     7286    /opt/software/linux-rhel7-x86_64/gcc-4.8.5/gcc-8.1.0-eesg3q2ggeqhvjpopgbpobxgyv6m5hg4/lib/gcc/x86_64-pc-linux-gnu/8.1.0/include/emmintrin.h:700:35: note: expected 'const __vector(2) l
             ong long int *' but argument is of type 'double *'
     7287     _mm_loadu_si128 (__m128i_u const *__P)
     7288                      ~~~~~~~~~~~~~~~~~^~~
     7289    /opt/spack/lib/spack/env/gcc/gcc -c -O2 -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DDYNAMIC_ARCH -DNO_WARMUP -DMAX_CPU_NUMBER=72 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SING
             LE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.12\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=dtrmm_iutncopy_SKYLAKEX -DASMFN
             AME=dtrmm_iutncopy_SKYLAKEX_ -DNAME=dtrmm_iutncopy_SKYLAKEX_ -DCNAME=dtrmm_iutncopy_SKYLAKEX -DCHAR_NAME=\"dtrmm_iutncopy_SKYLAKEX_\" -DCHAR_CNAME=\"dtrmm_iutncopy_SKYLAKEX\" -DNO_AFF
             INITY -DTS=_SKYLAKEX -I.. -DBUILD_KERNEL -DTABLE_NAME=gotoblas_SKYLAKEX -march=skylake-avx512 -DDOUBLE  -UCOMPLEX -Wno-uninitialized -DDOUBLE -UCOMPLEX -UOUTER -ULOWER -UUNIT generic/
             trmm_utcopy_16.c -o dtrmm_iutncopy_SKYLAKEX.o
     7290    /opt/spack/lib/spack/env/gcc/gcc -c -O2 -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DDYNAMIC_ARCH -DNO_WARMUP -DMAX_CPU_NUMBER=72 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SING
             LE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.12\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=dtrmm_iltucopy_SKYLAKEX -DASMFN
             AME=dtrmm_iltucopy_SKYLAKEX_ -DNAME=dtrmm_iltucopy_SKYLAKEX_ -DCNAME=dtrmm_iltucopy_SKYLAKEX -DCHAR_NAME=\"dtrmm_iltucopy_SKYLAKEX_\" -DCHAR_CNAME=\"dtrmm_iltucopy_SKYLAKEX\" -DNO_AFF
             INITY -DTS=_SKYLAKEX -I.. -DBUILD_KERNEL -DTABLE_NAME=gotoblas_SKYLAKEX -march=skylake-avx512 -DDOUBLE  -UCOMPLEX -Wno-uninitialized -DDOUBLE -UCOMPLEX -UOUTER -DLOWER -DUNIT generic/
             trmm_ltcopy_16.c -o dtrmm_iltucopy_SKYLAKEX.o
     7291    /opt/spack/lib/spack/env/gcc/gcc -c -O2 -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DDYNAMIC_ARCH -DNO_WARMUP -DMAX_CPU_NUMBER=72 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SING
             LE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.12\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=dtrmm_iltncopy_SKYLAKEX -DASMFN
             AME=dtrmm_iltncopy_SKYLAKEX_ -DNAME=dtrmm_iltncopy_SKYLAKEX_ -DCNAME=dtrmm_iltncopy_SKYLAKEX -DCHAR_NAME=\"dtrmm_iltncopy_SKYLAKEX_\" -DCHAR_CNAME=\"dtrmm_iltncopy_SKYLAKEX\" -DNO_AFF
             INITY -DTS=_SKYLAKEX -I.. -DBUILD_KERNEL -DTABLE_NAME=gotoblas_SKYLAKEX -march=skylake-avx512 -DDOUBLE  -UCOMPLEX -Wno-uninitialized -DDOUBLE -UCOMPLEX -UOUTER -DLOWER -UUNIT generic/
             trmm_ltcopy_16.c -o dtrmm_iltncopy_SKYLAKEX.o
  >> 7292    make[1]: *** [dasum_k_SKYLAKEX.o] Error 1
     7293    make[1]: *** Waiting for unfinished jobs....
     7294    make[1]: Leaving directory `/tmp/root/spack-stage/spack-stage-openblas-0.3.12-b45rhgos7s5yh5d6yivh43ucoirtcqcb/spack-src/kernel'
  >> 7295    make: *** [libs] Error 1
```

@siko1056 @t-karatsu @sethrj @mdorier @scottwittenburg @jabl

--- a/var/spack/repos/builtin/packages/openblas/package.py
+++ b/var/spack/repos/builtin/packages/openblas/package.py
@@ -112,6 +112,9 @@ class Openblas(MakefilePackage):
     patch('openblas_fujitsu_v0.3.11.patch', when='@0.3.11: %fj')
     patch('openblas_fujitsu2.patch', when='@0.3.10: %fj')
 
+    # See https://github.com/spack/spack/issues/19932
+    conflicts('%gcc@:8.2.99', when='@0.3.11:')
+
     # See https://github.com/spack/spack/issues/3036
     conflicts('%intel@16', when='@0.2.15:0.2.19')
     conflicts('+consistent_fpcsr', when='threads=none',

